module/zfs/zfs_vnops.c

```diff
@@ -1149,10 +1149,21 @@ zfs_get_data(void *arg, uint64_t gen, lr_write_t *lr, char *buf,
 	ASSERT3P(lwb, !=, NULL);
 	ASSERT3U(size, !=, 0);
 
+	error = zfs_zget_impl(zfsvfs, object, &zp, B_TRUE);
+#if defined(__linux__)
```

Comment:
I added ifdef because I'm not sure if it's possible to return EAGAIN in freebsd or not.
But I guess looking at the caller, it doesn't really matter if we return EAGAIN or EIO.

Reply:
perhaps it would be more clean to enhance `zil_lwb_commit`, which is the user of `zfs_get_data`, to recognize `EAGAIN`, I think